### DISCUSSION TITLE
Reuse semver selection when updating version

### DISCUSSION THREAD

**Discussion by alxndr** [vuki656/package-info.nvim#98] [Q&A] [answered]
NPM's dependency syntax allows us to [specify exact versions or ranges of versions to use](https://docs.npmjs.com/about-semantic-versioning) of dependencies.

I often specify exact versions in my `package.json`. I've found that this project's `change_version` function will prefix the new package version with the caret `^`, which I then remove. It would be great if this command would preserve the semver range specified with the existing version — i.e. if I have a version specified exactly (`"1.0.0"`), updating it to 1.0.1 would result in `"1.0.1"`.

(Not sure how to handle all cases with the tilde `~`, e.g. `"~1.0"` where the published version has gone from `1.0.1` to `1.1.0`... use `"~1.1"` I suppose?)

**Comment by vuki656** [comment, marked as answer]
There is actually no prefixing done by the plugin itself. 

It takes the name of the package you're on and runs `npm install name@version` (or with yarn) in the terminal. 

---

I just tried to remove the `^` from the version and then upgraded it to the latest version and `^` was put back next to the version so I think you shouldn't do that by hand. Might be wrong tho. 

Also, removing it by hand will make your `.lock` file out of date if you don't run `yarn/npm install` after it. I tested it by removing `^` from the version and then tried to update the package to the latest version by running `yarn upgrade name@latest` and I got the following 
`error Outdated lockfile. Please run yarn install and try again. info Visit https://yarnpkg.com/en/docs/cli/upgrade for documentation about this command.`

---

If there is a way to omit `^` from the version after version change that comes from `npm/yarn` CLI itself i'd be happy to implement it but I'm not a fan of manually stripping `^` from the version and then reinstalling the dependencies.
